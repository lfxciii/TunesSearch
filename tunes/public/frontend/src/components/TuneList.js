import React, { Component } from "react";

import {
    Card, Button, Image, Accordion
} from 'react-bootstrap';

// this component displays a list of tunes
export default class TuneList extends Component {

    render() {
        let tunes = [...this.props.tunes];
        let cards = null;

        if (tunes.length > 0) {
            cards = tunes.map((tune, index) => {
                return (
                    <Accordion defaultActiveKey="1">
                        <Card
                            id={index}
                            key={index}
                            className="m-1">
                            <Card.Header style={{
                                backgroundColor: "black",
                                opacity: "0.8"
                            }} as="h5">
                                <Accordion.Toggle as={Button} variant="link" eventKey="0">
                                    {tune.artistName} - {tune.trackName} ({new Date(tune.releaseDate).toLocaleDateString()})
                                </Accordion.Toggle>
                            </Card.Header>
                            <Accordion.Collapse eventKey="0">
                                <Card.Body style={{
                                    backgroundColor: "lightgray"
                                }}>
                                    <Image src={tune.artworkUrl60} />
                                    <Card.Title>{tune.collectionCensoredName}</Card.Title>
                                    <Card.Text>
                                        {tune.longDescription}
                                    </Card.Text>
                                    <Button className="btn btn-primary">Price ${tune.trackPrice}</Button>
                                    <Button href={tune.previewUrl} className="btn btn-warning m-1">Preview</Button>
                                    {
                                        (!tune.isFav) ?
                                            <Button
                                                id={tune.trackId}
                                                onClick={this.props.add}
                                                className="btn btn-success">+</Button>
                                            :
                                            <Button
                                                id={tune.trackId}
                                                onClick={this.props.remove}
                                                className="btn btn-danger">-</Button>
                                    }
                                </Card.Body>
                            </Accordion.Collapse>
                        </Card>
                    </Accordion>
                );
            });
        } else {
            cards = (
                <h4><em>No search results</em></h4>
            );
        }
        return (
            <div className="float-left w-75 d-inline-block rounded"
            >
                {cards}
            </div>
        );
    }
}



