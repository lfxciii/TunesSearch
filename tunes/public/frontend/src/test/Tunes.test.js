import React, { Component } from 'react';
let renderer = require('react-test-renderer');
// import Tunes from '../components/Tunes';
import TuneList from '../components/TuneList';

// getting funny errors here. not sure why
test("tunes render test", () => {
    const tree = renderer
        .create(<Tunes />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("tunes render test", () => {
    const tree = renderer
        .create(<TuneList 
            tunes={[{
                tarckId: 1,
                artistName: "Borderlands 3", 
                trackName: "best game ever",
                releaseDate: new Date().toString(),
                artworkUrl60: "",
                collectionCensoredName: "Hyperion",
                longDescription: "Awesome game",
                trackPrice: "0.00",
                previewUrl: ""                
            }]}
        />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});