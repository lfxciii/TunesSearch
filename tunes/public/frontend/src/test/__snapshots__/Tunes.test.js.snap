// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tunes render test 1`] = `
<div
  className="container rounded w-100"
>
  <p>
    sdfsdf
  </p>
  <div>
    <div
      className="float-left w-75 d-inline-block rounded"
    >
      <h4>
        <em>
          No search results
        </em>
      </h4>
    </div>
  </div>
  <div
    className="w-25 h-100 float-right rounded shadow"
  >
    <h2
      style={
        Object {
          "fontFamily": "Arial",
        }
      }
    >
      <strong>
        <i>
          Favourites
        </i>
      </strong>
    </h2>
    <p>
      <small>
        <strong>
          Add some favourites!
        </strong>
      </small>
    </p>
  </div>
</div>
`;
